atlp-rwanda/eagles-phantom-backend
# #O6QG2WdZ Reset the password

#### What does this PR do?

- This PR reset the password of a user who is in the system

#### Description of Task to be completed?

- create a controller for reset link

- create a validation for reset link

- create a router for reset link

#### How should this be manually tested?

- clone this repo

- run `npm i`

- Create a .env file and fill it with your own credentials taken example as in the .env.example 

- Create a SendGrid account and its key in the .env

- run `npm run migrate:reset`

- run `npm run seed`

- run `npm run dev`

- Create a SendGrid account and its key in the .env

- Test the endpoints in this documentation: `http://localhost:3020/api-docs/#/`

- After getting the link in the email, take the token and put it in the parameter of the reset-password,  and it starts right after `reset-password/`

- Make sure you put the token from admin login when token is asked in the forgot-link documentation

#### What are the relevant pivotal tracker stories?

- https://trello.com/c/O6QG2WdZ/22-reset-password

#### The screenshots related?

<img width="1440" alt="Screen Shot 2020-10-27 at 06 59 03" src="https://user-images.githubusercontent.com/44581262/97262325-45e08580-1829-11eb-8fcd-24673e5b183c.png">
<img width="1440" alt="Screen Shot 2020-10-27 at 06 59 12" src="https://user-images.githubusercontent.com/44581262/97262338-4d079380-1829-11eb-90ee-d5fa95121d3e.png">
<img width="1440" alt="Screen Shot 2020-10-27 at 06 59 25" src="https://user-images.githubusercontent.com/44581262/97262343-4ed15700-1829-11eb-8a6c-813b0e702a2b.png">
<img width="1440" alt="Screen Shot 2020-10-27 at 06 59 40" src="https://user-images.githubusercontent.com/44581262/97262349-5133b100-1829-11eb-9640-15b27b197798.png">
